tests/unit/scp_test.py

```diff
@@ -57,3 +58,45 @@ def test_get(dst_path, src_path, ssh_scp, transmit_payload):
     """Check that SCP file download works."""
     ssh_scp.get(str(src_path), str(dst_path))
     assert dst_path.read_bytes() == transmit_payload
+
+
+@pytest.fixture
+def dst_path_exists(tmp_path):
+    """Return a local path that exists."""
+    path = tmp_path / 'non-existing.txt'
+    path.write_bytes('whatever'.encode())
+    return path
+
+
+def test_get_existing_local(dst_path_exists, src_path, ssh_scp, transmit_payload):
+    """Check that SCP file download works and overrides local file if it exists."""
+    ssh_scp.get(str(src_path), str(dst_path_exists))
+    assert dst_path_exists.read_bytes() == transmit_payload
+
+
+@pytest.fixture
+def large_payload():
+    """Generate a large test payload."""
+    # buffer of random 1024 bytes -- just printable ones for easier debugging
+    rands = []
+    for _ in range(1024):
+        rnd = chr(random.randint(ord(' '), ord('~')))
+        rands.append(rnd)
+    rand = ''.join(rands)
+    # .. repeated 65 times
+    repeat = 65
+    return ''.join(rand for _ in range(repeat)).encode()
```

Comment:
I realized there's a lot of conversions between integers, bytes and text strings back and forth here. So here's some refactoring to make it better. I'll post a series of comments with iterations of my refactoring and then, will commit it myself. Stay tuned!

This is the first improvement:

```suggestion
@pytest.fixture
def large_payload():
    """Generate a large 64kB+1B test payload."""
    # buffer of random 1024 bytes -- just printable ones for easier debugging
    rands = []
    for _ in range(1024):
        rnd = chr(random.randint(ord(' '), ord('~')))
        rands.append(rnd)
    rand = ''.join(rands)

    full_bytes_number = 64
    return ''.join(rand for _ in range(full_bytes_number)).encode() + rand[0]
```

Reply:
```suggestion
@pytest.fixture
def large_payload():
    """Generate a large 64kB+1B test payload."""
    # buffer of random 1024 bytes -- just printable ones for easier debugging
    whitespace_position = ord(' ')
    tilda_position = ord('~')
    rands = []
    for _ in range(1024):
        rnd = chr(random.randint(whitespace_position, tilda_position))
        rands.append(rnd)
    rand = ''.join(rands)

    full_bytes_number = 64
    return ''.join(rand for _ in range(full_bytes_number)).encode() + rand[0]
```

Reply:
```suggestion
from itertools import chain


@pytest.fixture
def large_payload():
    """Generate a large 65537 byte (64kB+1B) test payload."""
    # buffer of random 1024 bytes -- just printable ones for easier debugging
    whitespace_position = ord(' ')
    tilda_position = ord('~')
    rands = [random.randint(whitespace_position, tilda_position) for _ in range(1024)]

    full_bytes_number = 64
    return bytes(chain.from_iterable(rands for _ in range(full_bytes_number))) + rands[0].to_bytes(1)
```

Reply:
```suggestion
@pytest.fixture
def large_payload():
    """Generate a large 65537 byte (64kB+1B) test payload."""
    # buffer of random 1024 bytes -- just printable ones for easier debugging
    whitespace_position = ord(' ')
    tilda_position = ord('~')
    rands = [random.randint(whitespace_position, tilda_position) for _ in range(1024)]

    full_bytes_number = 64
    return bytes(rands * full_bytes_number) + rands[0].to_bytes(1)
```

Reply:
```suggestion
import string


@pytest.fixture
def large_payload():
    """Generate a large 65537 byte (64kB+1B) test payload."""
    random_char_megabyte = [random.choice(string.printable) for _ in range(1024)]
    full_bytes_number = 64
    return bytes(random_char_megabyte * full_bytes_number) + random_char_megabyte[0].to_bytes(1)
```